llama2.mojo

```diff
@@ -492,18 +517,15 @@ fn rmsnorm(
     inout o: BufferPtrFloat32, x: BufferPtrFloat32, weight: BufferPtrFloat32, size: Int
 ) -> None:
     # Calculate sum of squares
-    var tmp = SIMD[DType.float32, nelts](0)
+    var tmp = Accumulator[DType.float32, nelts]()
 
     @parameter
     fn _sum2[_nelts: Int](j: Int):
-        if _nelts < nelts:
-            tmp[0] += (x.offset(j).simd_load[_nelts](0) ** 2).reduce_add()
-        else:
-            tmp += x.offset(j).simd_load[nelts](0) ** 2
+        tmp.accumulate(x.offset(j).simd_load[_nelts](0) ** 2)
```

Comment:
I think `nelts` not necessary == `_nelts` , so probably need to consider accumulation for tail elements. 
Or maybe `accumulate[_width: Int]` already handles this? If so that's cool feature

Reply:
This is the point of the PR.  By using a stack allocated pointer of size `nelts` we can always load and store with size `_nelts`.  `_nelts` is passed as a parameter so we can also pass it  to `Accumulator.accumulate` which can pass it to `DTypePointer.simd_store` and `DTypePointer.simd_load` so we can accumulate exactly `_nelts` size.

Reply:
And in this line we are relying on Mojo's new parameter inference to get `_nelts`.